### PR TITLE
switch the test suite to pytest

### PR DESCRIPTION
**Pull request by ev-br**
This scratches my itch of not having`nose` installed anymore.

### CODE REVIEW

**Comment by jnothman** [comment]
There are another couple of references to nose in the repo. Otherwise +1 from me

**Comment by ev-br** [comment]
I've found two references: one incantation in setup.py which is meant to enable `$ python setup.py test` --- is it still relevant? I tried to use instead the recommended pytest idiom using `pytest-runner` but did not manage to get it to work, and gave up. I suspect it's not worth it since `pytest --pyargs numpydoc` just works.

The other reference to nose, https://github.com/numpy/numpydoc/blob/master/numpydoc/numpydoc.py#L174
is in comments, and I've no idea if it's still relevant.

**Comment by mattip** [comment]
Now that numpy is using pytest in the soon-to-be-released 1.15, is it time to update and merge this? There is currently a merge conflict

**Comment by ev-br** [comment]
Rebased.

File: numpydoc/tests/test_docscrape.py [hunk 1]
```diff
@@ -167,52 +165,53 @@ def test_extended_summary():
...
+    assert arg_type == '(N, N) ndarray'
     assert desc[0].startswith('Covariance matrix')
     assert doc['Parameters'][0][-1][-1] == '   (1+2+3)/3'
 
 
 def test_other_parameters():
-    assert_equal(len(doc['Other Parameters']), 1)
-    assert_equal([n for n,_,_ in doc['Other Parameters']], ['spam'])
+    assert len(doc['Other Parameters']) == 1
+    assert [n for n,_,_ in doc['Other Parameters']] == ['spam']
```
[the remark this answers is not in the document]
Comment (ev-br) [reply numpydoc/tests/test_docscrape.py]: OK, fixed, here and elsewhere.

**Comment by jnothman** [comment]
Thanks!

**Pull request merged**